### ISSUE TITLE
[Network] Secure socket support ?

### ISSUE THREAD

**Issue by azjezz** [hhvm/hsl-experimental#102]
any plans to add TLS support to Network ? 

```hack
$server = await TCP\Server::createAsync(...);
$server = new Network\SecureServer($serve, $options);
```

**Comment by fredemmott** [comment]
Yep, definitely. Unclear if it'll be wrappers like that, or something else.

Likely won't be done before HSL IO has a new native implementation.

**Comment by fredemmott** [comment]
Now that native TCP is almost done, I've reached to FB's crypto people about the best way to go about doing this.

The 'obvious' way is to expose new FD-based bindings to `(open|libre|boring)ssl`, not sure if that's actually best practice nowadays.
